### DISCUSSION TITLE
Publishing failed :| Publishing to Microsoft Store failed: Microsoft Store is taking too long to start certification

### DISCUSSION THREAD

**Discussion by vadimbarda** [codemagic-ci-cd/codemagic-docs#2325] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

1 Publish application to Microsoft Store and see issue in log in Publishing section:
== Gathering artifacts ==

== Publishing artifacts ==

Publishing artifact release.zip
Publishing artifact media_casa.msix
Submission to Microsoft Store committed, waiting for certification phase
Publishing to Microsoft Store failed: Microsoft Store is taking too long to start certification

Build failed :|

Publishing failed :|
Publishing to Microsoft Store failed: Microsoft Store is taking too long to start certification
2 Check that application is published to Microsoft Store with no issues

### Expected results

Build succeeded

### Actual results

Build failed

### Build id (optional)

65ad559a623b168c222199bb

**Comment by icarusdust** [comment]
Hi @vadimbarda, could you please confirm if the error is false negative ? Do you see that `.msix` is uploaded to the Microsoft Partner Center regardless of the error ?

**Reply by vadimbarda** [reply]
Yes, `.msix` is uploaded to the Microsoft Partner Center and will be published automatically after verification

**Reply by icarusdust** [reply]
Thanks for confirming @vadimbarda. At that point, Codemagic uploaded the created `.msix ` binary to Microsoft Store and by the look of the error, Microsoft Store had some troubles with starting the certification process: 

`Publishing to Microsoft Store failed: Microsoft Store is taking too long to start certification`

In general, the certification process should take time, but this one took longer. I see that your subsequent upload was successful and certification was completed successfully.

**Reply by vadimbarda** [reply]
Yes, there are no issues with publishing in general. The issue is that build was marked as failed.

**Reply by icarusdust** [reply]
The submission is at the certification step in the Partner Center when the publishing step is being executed and at this point, Codemagic completes the MSIX submission successfully, but interestingly, Microsoft Store returns it a false negative. I will check this behaviour with Microsoft Store and keep you updated about the results.

**Reply by icarusdust** [reply]
As the Microsoft Partner Center has recently changed the interface slightly and some functionalities moved to a new place, connecting the Partner Center with Azure could have slightly changed. I am confirming with Microsoft if this behaviour could be related to it. I'll get back to you soon.

**Reply by icarusdust** [reply]
Hi @vadimbarda, looks like the instability with Microsoft Submission API was fixed. However, it would be great if you could join a call with me to explain how you configure your app's tenant and store IDs to link the Azure AD application with the Partner Center, that way we can check if you are missing something when configuring it. Here is [my link to](https://calendly.com/nihalagazade/codemagic-30) book a call.

**Comment by aitaro** [comment]
I also faced the exact same problem. build_id is 661216850ac330cce4625992.

After this build, a submission is generated in Microsoft Partner Center, but in the "Pricing and availability" section, it says "We can't show details for this section right now. Please refresh the page or try again later." error occurs.

I contacted Microsoft Partner Center support and got the following answer, but I didn't know how to fix it.
"I data fixed it to a different id to submission5. It appears that a new 'submission5' was re-created, triggering the bug in our service. I recommend advising the ISV to reuse the original 'submission5' to ensure continuity."

**Reply by icarusdust** [reply]
Thanks for your input @aitaro Please book a call from [this link](https://calendly.com/nihalagazade/codemagic-30?month=2024-04), so we can check it closely together.

**Comment by icarusdust** [comment]
I am snoozing this thread for now as it partially requires investigation from the Microsoft Partner Center side. In case of any further updates, I will let you know.

**Comment by jpohhhh** [comment]
+1 on this, it happens on all my Windows Store builds.

Given the persistence, I'm curious if there's another strategy than erroring the build.
ex. Is there a longer timeout CodeMagic can use for checking? 

Currently, it always displays as a build failure in CodeMagic. That might be for the best, but its worth noting that this can't be disabled via ignore_failure AFAIK.

**Reply by icarusdust** [reply]
Hi @jpohhhh, sadly it cannot be reproduced with my apps, so it is hard to say at what point the problem arises. Would you like to have a call and check this issue together, it'll be very helpful to see what you are doing differently in your project setup ? Ping me at nihal@codemagic.io please, so I can schedule a call at the earliest opportunity.

**Comment by lukemmtt** [comment]
+1 on this. I use the Codemagic workflow editor with publishing to the Microsoft Store. The workflow always succeeds in actually submitting the build for publishing (as is evident in the Partner Center), but Codemagic always reports "Microsoft Store is taking too long to start certification".

Besides the build error, which itself is a distracting false positive, it spends a long time (5 minutes, ~$0.25) in the publishing step before throwing this error.

**Comment by icarusdust** [comment]
@lukemmtt please share one of your latest builds, will have a fresh look into it. You can find a build ID under the Build overview section in the build page.

**Reply by lukemmtt** [reply]
Thanks Nihal, you can look to build ID 67dddaf48c011d81f4fa63c0

**Reply by lukemmtt** [reply]
For what it's worth, the first build in a while has just actually _succeeded_—(I think I've seen 10+ builds with the failure). 

The build that succeeded was build ID 67ddef5e49162cd7b1652620.

Of course, I'd be grateful if the issue could still be investigated.

**Reply by icarusdust** [reply]
Thanks for sharing it, I will continue investigating it, will keep you updated with my findings.

**Reply by lukemmtt** [reply]
Great, thank you!

**Reply by lukemmtt** [reply]
Got another build that failed for the same reason; build ID: 67e1ba6ec78a8cdbe4fa4e84

**Reply by icarusdust** [reply]
Thanks, this helps. The team is trying to reproduce it @lukemmtt and see where this comes from, I will keep you updated.

**Reply by lukemmtt** [reply]
Sounds good, thank you! I am happy to help in any way I can. All I can say at this point is that despite the "failure" & the wasted build minutes, I can confirm that it is indeed building and publishing just fine—see screenshot:
 
<img width="384" alt="Screenshot 2025-03-24 at 4 36 26 PM" src="https://github.com/user-attachments/assets/9f3387de-927f-4e99-b010-0b702283fb2e" />

**Reply by icarusdust** [reply]
thanks, will figure out if this comes from the Microsoft API or where and post about the results. I will check your used minutes, please do not worry about them.

**Reply by lukemmtt** [reply]
Great, and thank you so much for the consideration with the minutes, that’s very kind!

**Reply by icarusdust** [reply]
@lukemmtt I see David from the support team is trying to help you with this issue in Intercom. He is trying to reproduce it and will need your feedback to him in your existing Intercom thread.

**Reply by lukemmtt** [reply]
I will do that, thank you 🙏🏻
